pytorch_pfn_extras/training/extension.py

```diff
@@ -222,3 +224,60 @@ def decorator(ext: 'ExtensionLike') -> 'ExtensionLike':
 
 def _as_extension(ext: 'ExtensionLike') -> Extension:
     return ext if isinstance(ext, Extension) else _WrappedExtension(ext)
+
+
+class ExtensionEntry:
+    """Extension and options.
+
+    Args:
+        extension: An extension.
+        name: Name of extension. If omitted, copied from the attributes of
+            ``extension`` argument.
+        priority: Invocation priority of the extension.
+            Default is ``PRIORITY_READER``.
+        trigger: Trigger object that determines when to invoke the extension.
+            Default is ``(1, 'iteration')``.
+        call_before_training: Flag to call extension before training.
+            Default is ``False``.
+
+    .. seealso::
+       :meth:`pytorch_pfn_extras.training.ExtensionsManager.extend`
+    """
+
+    def __init__(
+            self,
+            extension: 'ExtensionLike',
+            *,
+            name: Optional[str] = None,
+            priority: Optional[int] = None,
+            trigger: Optional['TriggerLike'] = None,
+            call_before_training: bool = False,
+    ) -> None:
+        self.extension = _as_extension(extension)
+        self.priority = priority or self.extension.priority
+        self.call_before_training = call_before_training
+
+        self.set_trigger(trigger or self.extension.trigger)
+        self.set_name(name or self.extension.name or self.extension.default_name)
+
+    def set_trigger(self, trigger: 'TriggerLike') -> None:
```

Comment:
Please make it private as this is a public class.
```suggestion
    def _update_trigger(self, trigger: 'TriggerLike') -> None:
```